### ISSUE TITLE
Empty edits sent to analysis server

### ISSUE THREAD

**Issue by DanTup** [Dart-Code/Dart-Code#587]
We sometimes seem to be sending these:

```js
{
	"id": "11",
	"method": "analysis.updateContent",
	"params": {
		"files": {
			"/Users/danny/Dev/Google/flutter/examples/hello_world/lib/main.dart": {
				"edits": [
					{
						"offset": 0,
						"length": 0,
						"replacement": "",
						"id": ""
					}
				],
				"type": "change"
			}
		}
	}
}
```

**Comment by DanTup** [comment]
This turned out to be a workaround (see https://github.com/dart-lang/sdk/issues/30238). We can probably remove this soon, depending on when the fix shipped.

**Comment by DanTup** [comment]
Fixed in 5b338d1566523f5371e3292a5bc3cf6c380aabb8.
